apps/synapse-oauth-signin/src/OAuth2Form.tsx

```diff
@@ -60,248 +73,192 @@ export function OAuth2Form() {
-            const requestObject = getURLParam('request')
-            const requestUri = getURLParam('request_uri')
-            if (requestObject) {
-              handleErrorRedirect(new OAuthClientError('request_not_supported'))
-            }
-            if (requestUri) {
-              handleErrorRedirect(
-                new OAuthClientError('request_uri_not_supported'),
-              )
-            }
-            // sorry, we don't support registration (yet?)
-            // https://openid.net/specs/openid-connect-core-1_0.html#RegistrationParameter
-            const registration = getURLParam('registration')
-            if (registration) {
-              handleErrorRedirect(
-                new OAuthClientError('registration_not_supported'),
-              )
-            }
-          }
-        })
-        .catch(_err => {
-          // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
-          onError(_err)
-        })
-    }
+  const onDeny = () => {
+    sendGTagEvent('UserDeniedConsent')
+    // Redirect with 'access_denied' per https://datatracker.ietf.org/doc/html/rfc6749#section-4.1.2.1
+    handleErrorRedirect(new OAuthClientError('access_denied'))
   }
 
-  const getOIDCAuthorizationRequestFromSearchParams = () => {
-    const claimsString: string = getURLParam('claims')!
-    const authRequest: OIDCAuthorizationRequest = {
-      clientId: getURLParam('client_id')!,
-      scope: getURLParam('scope')!,
-      responseType: getURLParam('response_type')!,
-      redirectUri: getURLParam('redirect_uri')!,
-      nonce: getURLParam('nonce'),
-    }
-    if (claimsString) {
-      authRequest.claims = JSON.parse(claimsString)
-    }
-    return authRequest
-  }
+  const { data: oidcRequestDescription, error: oidcRequestDescriptionError } =
+    SynapseQueries.useGetOAuth2RequestDescription(
+      oidcAuthorizationRequestFromSearchParams!,
+      {
+        enabled: Boolean(oidcAuthorizationRequestFromSearchParams),
+      },
+    )
 
-  //init
   useEffect(() => {
-    const getOauthClientInfo = () => {
-      if (!oauthClientInfo && !error) {
-        const code = getURLParam('code')
-        if (code) return // we're in the middle of a SSO, do not attempt to get OAuthClient info yet
+    if (oidcRequestDescription) {
+      sendGTagEvent('SynapseOAuthClientRequestDescriptionLoaded')
 
-        const clientId = getURLParam('client_id')
-        if (!clientId) {
-          onError(new Error('Synapse OAuth client_id is required'))
-          return
-        }
-        SynapseClient.getOAuth2Client(clientId)
-          .then((oauthClientInfo: OAuthClientPublic) => {
-            if (oauthClientInfo.verified) {
-              getOAuth2RequestDescription()
-            }
-            setOauthClientInfo(oauthClientInfo)
-          })
-          .catch(_err => {
-            // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
-            onError(_err)
-          })
```

Comment:
This behavior (basically, check if we're in the middle of SSO via Google/ORCiD before we try to do anything else) was moved up higher.